src/runtime_functions.js

```diff
@@ -158,7 +168,7 @@ function addFunctionWasm(func, sig) {
     table.set(ret, wrapped);
   }
 
-  functionsInTableMap[func] = ret;
```

Comment:
Whats the difference between these two lines?

Reply:
`functionsInTableMap[func]` treats it as a JS object. but that isn't the right way to use a WeakMap, which has special functions. It's a pitfall since in JS it's also an object, so the notation is valid and does not throw, but doesn't actually assign to the WeakMap (it assigns a JS object property... which roots strongly, and isn't found later)